### PR TITLE
   fix multiple system unconfigured rise cleanup

### PR DESCRIPTION
**Pull request by anscipione**
#1171

### CODE REVIEW

**Comment by anscipione** [comment]
> Hi @anscipione, thanks for your PR and reporting the issue. Have you checked the test locally? This change breaks the tests.
> 
> I have a proposal to make the change more readable. Please add these lines at the very beginning of the methods.
> 
> ```
> if (impl_->get_state().id() == lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED)
> {
>   return return_type::OK;
> }
> ```
> 
> And we should update the tests too.

I applied the suggested change.
I also fixed the tests, but I would prefer you to check them.

**Review by destogl — APPROVED** [review]

**Review by saikishor — APPROVED** [review]

File: hardware_interface/src/actuator.cpp [hunk 1]
```diff
@@ -220,6 +220,11 @@ return_type Actuator::read(const rclcpp::Time & time, const rclcpp::Duration & p
 {
   // TODO(destogl): discuss what should be default return value, e.g., "NOT_EXECUTED"
```
Comment (destogl) [inline hardware_interface/src/actuator.cpp:221]:
```suggestion
  if (impl_->get_state().id() == lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED)
  {
    return return_type::OK;
  }
```

File: hardware_interface/src/actuator.cpp [hunk 2]
```diff
@@ -220,6 +220,11 @@ return_type Actuator::read(const rclcpp::Time & time, const rclcpp::Duration & p
...
   return_type result = return_type::ERROR;
+  if (impl_->get_state().id() == lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED)
+  {
+    return return_type::OK;
+  }
+
```
Comment (destogl) [inline hardware_interface/src/actuator.cpp:227]:
```suggestion
```

File: hardware_interface/src/actuator.cpp [hunk 3]
```diff
@@ -237,6 +242,11 @@ return_type Actuator::write(const rclcpp::Time & time, const rclcpp::Duration &
 {
   // TODO(destogl): discuss what should be default return value, e.g., "NOT_EXECUTED"
```
Comment (destogl) [inline hardware_interface/src/actuator.cpp:243]:
```suggestion
  if (impl_->get_state().id() == lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED)
  {
    return return_type::OK;
  }
```

File: hardware_interface/src/actuator.cpp [hunk 4]
```diff
@@ -237,6 +242,11 @@ return_type Actuator::write(const rclcpp::Time & time, const rclcpp::Duration &
...
   return_type result = return_type::ERROR;
+  if (impl_->get_state().id() == lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED)
+  {
+    return return_type::OK;
+  }
+
```
Comment (destogl) [inline hardware_interface/src/actuator.cpp:249]:
```suggestion
```

File: hardware_interface/src/sensor.cpp [hunk 5]
```diff
@@ -197,6 +197,11 @@ return_type Sensor::read(const rclcpp::Time & time, const rclcpp::Duration & per
 {
   // TODO(destogl): discuss what should be default return value, e.g., "NOT_EXECUTED"
```
Comment (destogl) [inline hardware_interface/src/sensor.cpp:198]:
```suggestion
  if (impl_->get_state().id() == lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED)
  {
    return return_type::OK;
  }
```

File: hardware_interface/src/sensor.cpp [hunk 6]
```diff
@@ -197,6 +197,11 @@ return_type Sensor::read(const rclcpp::Time & time, const rclcpp::Duration & per
...
   return_type result = return_type::ERROR;
+  if (impl_->get_state().id() == lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED)
+  {
+    return return_type::OK;
+  }
+
```
Comment (destogl) [inline hardware_interface/src/sensor.cpp:204]:
```suggestion
```

File: hardware_interface/src/system.cpp [hunk 7]
```diff
@@ -216,6 +216,11 @@ return_type System::read(const rclcpp::Time & time, const rclcpp::Duration & per
 {
   // TODO(destogl): discuss what should be default return value, e.g., "NOT_EXECUTED"
```
Comment (destogl) [inline hardware_interface/src/system.cpp:217]:
```suggestion
  if (impl_->get_state().id() == lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED)
  {
    return return_type::OK;
  }
```

File: hardware_interface/src/system.cpp [hunk 8]
```diff
@@ -216,6 +216,11 @@ return_type System::read(const rclcpp::Time & time, const rclcpp::Duration & per
...
   return_type result = return_type::ERROR;
+  if (impl_->get_state().id() == lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED)
+  {
+    return return_type::OK;
+  }
+
```
Comment (destogl) [inline hardware_interface/src/system.cpp:223]:
```suggestion
```

File: hardware_interface/src/system.cpp [hunk 9]
```diff
@@ -233,6 +238,11 @@ return_type System::write(const rclcpp::Time & time, const rclcpp::Duration & pe
 {
   // TODO(destogl): discuss what should be default return value, e.g., "NOT_EXECUTED"
```
Comment (destogl) [inline hardware_interface/src/system.cpp:239]:
```suggestion
  if (impl_->get_state().id() == lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED)
  {
    return return_type::OK;
  }
```

File: hardware_interface/src/system.cpp [hunk 10]
```diff
@@ -233,6 +238,11 @@ return_type System::write(const rclcpp::Time & time, const rclcpp::Duration & pe
...
   return_type result = return_type::ERROR;
+  if (impl_->get_state().id() == lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED)
+  {
+    return return_type::OK;
+  }
+
```
Comment (destogl) [inline hardware_interface/src/system.cpp:245]:
```suggestion
```

**Review by destogl — COMMENTED** [review]
We have to move these methods as in the proposals. I cannot merge those directly, unfortunately.

**Comment by saikishor** [comment]
Hello @anscipione!

Do you have time to address the comments of @destogl? or shall i take over and fix it in a different PR?. Right now, I ended up doing the same without knowing about this PR

Thank you,

**Comment by anscipione** [comment]
I'll fix asap.

**Comment by destogl** [comment]
I couldn't introduce changes to this, so opened a follow-up #1279

**Pull request closed without merging**